mail/opensmtpd/Portfile

```
@@ -0,0 +1,114 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+
+github.setup        opensmtpd opensmtpd 6.8.0p1-rc1
+categories          mail
+platforms           darwin
+license             ISC
+maintainers         debian.org:rak
+description         secure, reliable, lean, and easy-to configure SMTP server
+long_description    The OpenSMTPD server seeks to be \
+                    \n* as secure as possible, and uses privilege separation to mitigate \
+                    \n  possible security bugs \
+                    \n* as reliable as possible: any accepted email must not be lost \
+                    \n* lean: it covers typical usage cases instead of every obscure one \
+                    \n* easy to configure, with a configuration syntax reminiscent of the OpenBSD \
+                    \n  Packet Filter's (PF) \
+                    \n* fast and efficient: it can handle large queues with reasonable performance
+homepage            https://www.opensmtpd.org
+
+github.tarball_from archive
+
+checksums           rmd160  186ee26a72f63aa3058bf9990a5a9db12f8ac2ae \
+                    sha256  e35a2954dad002ffaf2060358aa97e5268dbe653e22cdd5ebf7addb1dcabdcdb \
+                    size    447879
+
+depends_build       port:autoconf \
+                    port:automake \
+                    port:bison \
+                    port:libtool
+
+depends_lib         port:libevent \
+                    port:openssl
+
+startupitem.create  yes
+startupitem.name    OpenSMTPD
+startupitem.autostart   yes
+startupitem.executable  ${prefix}/sbin/smtpd
+
+add_users           _smtpd group=_smtpd realname=OpenSMTP\ Daemon \
+                    _smtpq group=_smtpq realname=OpenSMTP\ Queue
+
+set sysconfdir ${prefix}/etc/${name}
+
+post-extract {
+    reinplace "s|file:/etc/mail/aliases|file:${sysconfdir}/aliases|g" ${worksrcpath}/usr.sbin/smtpd/smtpd.conf
+}
+
+#use_autoreconf      yes
+#autoreconf.cmd      ./bootstrap
+
+pre-configure {
+    system -W ${worksrcpath} "./bootstrap"
+}
+
+configure.args      --with-auth-pam \
+                    --with-libevent=${prefix} \
+                    --sysconfdir=${sysconfdir} \
+                    --with-path-queue=${prefix}/var/spool/smtpd \
+                    --with-table-db
+
+destroot.keepdirs   ${destroot}${prefix}/var/spool/smtpd
+
+pre-destroot {
+    addgroup _smtpd
+    addgroup _smtpq
+}
+
+post-destroot {
+    xinstall -o root -g wheel -m 711 -d ${destroot}${prefix}/var/spool/smtpd
```

This will prevent the port from being installed if the user has not installed MacPorts as root.
```suggestion
    xinstall -o ${install.user} -g ${install.group} -m 0711 -d ${destroot}${prefix}/var/spool/smtpd
```

---

OpenSMTPD checks that the directory is owned by uid 0 and gid 0 with permissions 711. Here's the [check](https://github.com/OpenSMTPD/OpenSMTPD/blob/master/usr.sbin/smtpd/queue_backend.c#L143), where `ckdir` is defined [here](https://github.com/OpenSMTPD/OpenSMTPD/blob/master/usr.sbin/smtpd/util.c#L189).

What's the appropriate solution? Add `destroot.asroot yes`?